jetbrains-core/src/software/aws/toolkits/jetbrains/services/schemas/search/SchemaSearchDialog.kt

```diff
@@ -4,48 +4,39 @@
 package software.aws.toolkits.jetbrains.services.schemas.search
 
 import com.intellij.openapi.project.Project
-import com.intellij.util.Alarm
+import com.intellij.ui.components.JBLabel
+import com.intellij.util.NotNullFunction
 import software.aws.toolkits.jetbrains.services.schemas.SchemaViewer
 import software.aws.toolkits.resources.message
 import software.aws.toolkits.telemetry.Result
 import software.aws.toolkits.telemetry.SchemasTelemetry
-import java.util.concurrent.CompletionStage
-import java.util.stream.IntStream
-import kotlin.streams.toList
+import javax.swing.JComponent
 
-interface SchemaSearchDialog<T : SchemaSearchResultBase, U : SchemaSearchDialogState<T>> {
+interface SchemaSearchDialog {
     fun initializeNew()
-    fun initializeFromState(state: U)
+    fun initializeFromState(state: SchemaSearchDialogState)
 }
 
 class SchemaSearchSingleRegistryDialog(
     private val registryName: String,
     project: Project,
-    onCancelCallback: (SchemaSearchSingleRegistryDialogState) -> Unit,
+    onCancelCallback: (SchemaSearchDialogState) -> Unit,
     private val searchExecutor: SchemaSearchExecutor = SchemaSearchExecutor(project),
-    schemaViewer: SchemaViewer = SchemaViewer(project),
-    alarmThreadToUse: Alarm.ThreadToUse = Alarm.ThreadToUse.SWING_THREAD
-) : SchemasSearchDialogBase<SchemaSearchResult, SchemaSearchSingleRegistryDialogState>(
+    schemaViewer: SchemaViewer = SchemaViewer(project)
+) : SchemasSearchDialogBase(
     project,
     schemaViewer,
     message("schemas.search.header.text.singleRegistry", registryName),
-    onCancelCallback,
-    alarmThreadToUse
+    onCancelCallback
 ) {
 
-    override fun getCurrentState(): SchemaSearchSingleRegistryDialogState {
-        val searchResults = IntStream.range(0, resultsModel.size()).mapToObj(resultsModel::get).toList()
-        return SchemaSearchSingleRegistryDialogState(currentSearchText(), searchResults, selectedSchema(), selectedSchemaVersion()?.version)
+    override fun createResultRenderer(): NotNullFunction<SchemaSearchResultWithRegistry, JComponent> = NotNullFunction {
```

Comment:
is the NotNullFunction required in the base?

Reply:
Cleaned this up and used Kotlin Lambda